test failure from IP reservation that was not yet deleted

It appears that when deleting IP reservations, it may be necessary to wait for the reservation to be deleted before other operations continue.

```
=== RUN   TestAccMetalReservedIPBlock_Public
2021/07/23 13:41:32 [DEBUG] POST https://api.equinix.com/metal/v1/projects
2021/07/23 13:41:32 [DEBUG] GET https://api.equinix.com/metal/v1/projects/6287dc83-7a3e-44bc-9e19-b15f7261815f
2021/07/23 13:41:33 [DEBUG] GET https://api.equinix.com/metal/v1/projects/6287dc83-7a3e-44bc-9e19-b15f7261815f/bgp-config
2021/07/23 13:41:33 [DEBUG] POST https://api.equinix.com/metal/v1/projects/6287dc83-7a3e-44bc-9e19-b15f7261815f/ips
2021/07/23 13:41:34 [DEBUG] GET https://api.equinix.com/metal/v1/ips/11caa108-16fd-47a3-adc3-fc73bdd0f7cd
    resource_metal_reserved_ip_block_test.go:90: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # metal_reserved_ip_block.test must be replaced
        -/+ resource "metal_reserved_ip_block" "test" {
              ~ address        = "147.75.105.1" -> (known after apply)
              ~ address_family = 4 -> (known after apply)
              ~ cidr           = 31 -> (known after apply)
              ~ cidr_notation  = "147.75.105.0/31" -> (known after apply)
              ~ gateway        = "147.75.105.0" -> (known after apply)
              ~ global         = false -> (known after apply)
              ~ id             = "11caa108-16fd-47a3-adc3-fc73bdd0f7cd" -> (known after apply)
              ~ manageable     = true -> (known after apply)
              ~ management     = false -> (known after apply)
              - metro          = "ny" -> null # forces replacement
              ~ netmask        = "255.255.255.254" -> (known after apply)
              ~ network        = "147.75.105.0" -> (known after apply)
              ~ public         = true -> (known after apply)
                tags           = [
                    "Tag1",
                    "Tag2",
                ]
                # (4 unchanged attributes hidden)
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
2021/07/23 13:41:35 [DEBUG] DELETE https://api.equinix.com/metal/v1/ips/11caa108-16fd-47a3-adc3-fc73bdd0f7cd
2021/07/23 13:41:35 [DEBUG] DELETE https://api.equinix.com/metal/v1/projects/6287dc83-7a3e-44bc-9e19-b15f7261815f
2021/07/23 13:41:35 [DEBUG] GET https://api.equinix.com/metal/v1/ips/11caa108-16fd-47a3-adc3-fc73bdd0f7cd
--- FAIL: TestAccMetalReservedIPBlock_Public (5.27s)
```

--- a/metal/resource_metal_reserved_ip_block.go
+++ b/metal/resource_metal_reserved_ip_block.go
@@ -100,17 +100,26 @@ func resourceMetalReservedIPBlock() *schema.Resource {
 		ForceNew:      true,
 		ConflictsWith: []string{"facility"},
 		Description:   "Metro where to allocate the public IP address block, makes sense only for type==public_ipv4, must be empty for type==global_ipv4, conflicts with facility",
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+		DiffSuppressFunc: func(k, fromState, fromHCL string, d *schema.ResourceData) bool {
 			_, facOk := d.GetOk("facility")
-			// new - new val from template
-			// old - old val from state
-			//
-			// suppress diff if metro is manually set for first time, and
-			// facility is already set
-			if len(new) > 0 && old == "" && facOk {
-				return facOk
+
+			// if facility is not in state, treat the diff normally, otherwise do following messy checks:
+			if facOk {
+				// If metro from HCL is specified, but not present in state, supress the diff.
+				// This is legacy, and I think it's here because of migration, so that old
+				// facility reservations are not recreated when metro is specified ???)
+				if fromHCL != "" && fromState == "" {
+					return true
+				}
+				// If metro is present in state but not present in HCL, supress the diff.
+				// This is for "facility-specified" reservation blocks created after ~July 2021.
+				// These blocks will have metro "computed" to the TF state, and we don't want to
+				// emit a diff if the metro field is empty in HCL.
+				if fromHCL == "" && fromState != "" {
+					return true
+				}
 			}
-			return old == new
+			return fromState == fromHCL
 		},
 		StateFunc: toLower,
 	}

--- a/metal/resource_metal_reserved_ip_block_test.go
+++ b/metal/resource_metal_reserved_ip_block_test.go
@@ -17,10 +17,10 @@ resource "metal_project" "foobar" {
 }
 
 resource "metal_reserved_ip_block" "test" {
-	project_id = "${metal_project.foobar.id}"
-	type     = "global_ipv4"
+	project_id  = metal_project.foobar.id
+	type        = "global_ipv4"
 	description = "testdesc"
-	quantity = 1
+	quantity    = 1
 }`, name)
 }
 
@@ -31,7 +31,7 @@ resource "metal_project" "foobar" {
 }
 
 resource "metal_reserved_ip_block" "test" {
-	project_id  = "${metal_project.foobar.id}"
+	project_id  = metal_project.foobar.id
 	facility    = "ewr1"
 	type        = "public_ipv4"
 	quantity    = 2
@@ -46,7 +46,7 @@ resource "metal_project" "foobar" {
 }
 
 resource "metal_reserved_ip_block" "test" {
-	project_id  = "${metal_project.foobar.id}"
+	project_id  = metal_project.foobar.id
 	metro       = "sv"
 	type        = "public_ipv4"
 	quantity    = 2
@@ -135,7 +135,7 @@ func TestAccMetalReservedIPBlock_Metro(t *testing.T) {
 	})
 }
 
-func TestAccMetalReservedIPBlock_importBasic(t *testing.T) {
+func TestAccMetalReservedIPBlock_ImportBasic(t *testing.T) {
 
 	rs := acctest.RandString(10)
 
@@ -171,6 +171,49 @@ func testAccCheckMetalReservedIPBlockDestroy(s *terraform.State) error {
 	return nil
 }
 
+func testAccCheckMetalReservedIPBlockConfig_FacilityToMetro(line string) string {
+	return fmt.Sprintf(`
+resource "metal_project" "foobar" {
+	name = "tfacc-reserved_ip_block_fac_met_test"
+}
+
+resource "metal_reserved_ip_block" "test" {
+	project_id  = metal_project.foobar.id
+	%s
+	type        = "public_ipv4"
+	quantity    = 2
+	tags        = ["Tag1", "Tag2"]
+}`, line)
+}
+
+func TestAccMetalReservedIPBlock_FacilityToMetro(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMetalReservedIPBlockDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckMetalReservedIPBlockConfig_FacilityToMetro(`   facility = "ny5"`),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"metal_reserved_ip_block.test", "facility", "ny5"),
+					resource.TestCheckResourceAttr(
+						"metal_reserved_ip_block.test", "metro", "ny"),
+				),
+			},
+			{
+				Config: testAccCheckMetalReservedIPBlockConfig_FacilityToMetro(`   metro = "ny"`),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"metal_reserved_ip_block.test", "metro", "ny"),
+				),
+				PlanOnly: true,
+			},
+		},
+	})
+}
+
 func testAccMetalReservedIP_Device(name string) string {
 	return fmt.Sprintf(`
 resource "metal_project" "foobar" {
@@ -203,7 +246,7 @@ resource "metal_device" "test" {
 `, name)
 }
 
-func TestAccMetalReservedIPDevice(t *testing.T) {
+func TestAccMetalReservedIPBlock_Device(t *testing.T) {
 
 	rs := acctest.RandString(10)
 

